modules/tensor_mechanics/tests/crystal_plasticity/crysp_substep.i

```diff
@@ -0,0 +1,229 @@
+  [../]
+  [./e_zz]
+    type = RankTwoAux
+    variable = e_zz
+    rank_two_tensor = lage
+    index_j = 2
+    index_i = 2
+    execute_on = timestep_end
+    block = 0
+  [../]
+  [./rotout]
+    type = CrystalPlasticityRotationOutAux
+    variable = rotout
+    execute_on = timestep_end
+    block = 0
+  [../]
+  [./gss1]
+    type = CrystalPlasticitySlipSysAux
+    variable = gss1
+    slipsysvar = gss
+    index_i = 1
+    execute_on = timestep_end
+    block = 0
+  [../]
+[]
+
+[BCs]
+  [./symmy]
+    type = PresetBC
+    variable = uy
+    boundary = bottom
+    value = 0
+  [../]
+  [./symmx]
+    type = PresetBC
+    variable = ux
+    boundary = left
+    value = 0
+  [../]
+  [./symmz]
+    type = PresetBC
+    variable = uz
+    boundary = back
+    value = 0
+  [../]
+  [./tdisp]
+    type = FunctionPresetBC
+    variable = uz
+    boundary = front
+    function = tdisp
+  [../]
+[]
+
+[Materials]
+  active = 'crysp'
+  [./crysp]
+    type = FiniteStrainCrystalPlasticity
+    block = 0
+    disp_y = uy
+    disp_x = ux
+    gtol = 1e-2
+    slip_sys_file_name = input_slip_sys.txt
+    disp_z = uz
+    C_ijkl = '1.684e5 1.214e5 1.214e5 1.684e5 1.214e5 1.684e5 0.754e5 0.754e5 0.754e5'
+    nss = 12
+    num_slip_sys_flowrate_props = 2 #Number of properties in a slip system
+    flowprops = '1 4 0.001 0.1 5 8 0.001 0.1 9 12 0.001 0.1'
+    hprops = '1.0 541.5 60.8 109.8 2.5'
+    gprops = '1 4 60.8 5 8 60.8 9 12 60.8'
+    fill_method = symmetric9
+    tan_mod_type = exact
+    gen_random_stress_flag = false;
+    maximum_substep_iteration = 2
+  [../]
+  [./elastic]
+    type = FiniteStrainElasticMaterial
+    block = 0
+    disp_y = uy
+    disp_x = ux
+    disp_z = uz
+    C_ijkl = '1.684e5 1.214e5 1.214e5 1.684e5 1.214e5 1.684e5 0.754e5 0.754e5 0.754e5'
+    fill_method = symmetric9
+  [../]
+[]
+
+[Postprocessors]
+  [./stress_zz]
+    type = ElementAverageValue
+    variable = stress_zz
+    block = 'ANY_BLOCK_ID 0'
+  [../]
+  [./fp_zz]
+    type = ElementAverageValue
+    variable = fp_zz
+    block = 'ANY_BLOCK_ID 0'
+  [../]
+  [./e_zz]
+    type = ElementAverageValue
+    variable = e_zz
+    block = 'ANY_BLOCK_ID 0'
+  [../]
+  [./gss1]
+    type = ElementAverageValue
+    variable = gss1
+    block = 'ANY_BLOCK_ID 0'
+  [../]
+[]
+
+[Preconditioning]
+  [./smp]
+    type = SMP
+    full = true
+  [../]
+[]
+
+[Executioner]
+  type = Transient
+  dt = 0.5
+
+  #Preconditioned JFNK (default)
+  solve_type = 'PJFNK'
+
+  petsc_options_iname = -pc_hypre_type
+  petsc_options_value = boomerang
+  nl_abs_tol = 1e-10
+  nl_rel_step_tol = 1e-10
+  dtmax = 10.0
+  nl_rel_tol = 1e-10
+  ss_check_tol = 1e-10
+  end_time = 30.0
+  dtmin = 0.5
+  num_steps = 3
+  nl_abs_step_tol = 1e-10
+[]
+
+[Outputs]
+  file_base = crysp_substep_out
+  output_initial = true
+  exodus = true
+  csv = true
+  gnuplot = true
+  print_linear_residuals = true
+  print_perf_log = true
+[]
+
+[Problem]
+  use_legacy_uo_initialization = false
```

Comment:
None of the modules have legacy flags turned on anymore. This parameter is no longer necessary. It's fine having it in there but you can safely leave it off for future moose or modules tests. You may still need it in Marmot though until Marmot is cleaned up.